src/NetAnalyzers/CSharp/Microsoft.NetCore.Analyzers/Performance/CSharpPreferIsEmptyOverCount.Fixer.cs

```diff
@@ -15,26 +15,62 @@ namespace Microsoft.NetCore.CSharp.Analyzers.Performance
     [ExportCodeFixProvider(LanguageNames.CSharp), Shared]
     public sealed class CSharpPreferIsEmptyOverCountFixer : PreferIsEmptyOverCountFixer
     {
-        protected override SyntaxNode? GetExpressionFromMemberAccessor(SyntaxNode node)
+        protected override SyntaxNode? GetObjectExpressionFromOperation(SyntaxNode node, string operationKey)
         {
-            while (node is ParenthesizedExpressionSyntax parenthesizedExpression)
+            SyntaxNode? countNode = null;
+            switch (operationKey)
```

Comment:
What about using the new switch pattern here?

Reply:
We don't use one-liners or single calls here to determine `countNode`, I don't see it too beneficial. 

Or perhaps I don't see it the way you do, can you provide a snippet?

Reply:
I was thinking about something like

```csharp
var countNode = operationKey switch
{
	UseCountProperlyAnalyzer.OperationBinaryLeft when (node is BinaryExpressionSyntax binary) => binary.Left,
    UseCountProperlyAnalyzer.OperationBinaryRight when (node is BinaryExpressionSyntax binary) => 	binary.Right,
   UseCountProperlyAnalyzer.OperationEqualsArgument when (node is InvocationExpressionSyntax invocation) => invocationExpression.ArgumentList.Arguments[0].Expression,
	UseCountProperlyAnalyzer.OperationEqualsInstance when (node is InvocationExpressionSyntax invocation && invocation.Expression is MemberAcessExpressionSyntax memberAccess) => memberAccess.Expression,
	_ => // Not sure what you want to do here
};
```